sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -75,6 +76,11 @@ class SQLContext(@transient val sparkContext: SparkContext)
    */
   protected[sql] def conf = currentSession().conf
 
+  // `listener` should be only used in the driver
+  @transient private[sql] val listener = new SQLListener(this)
+  sparkContext.addSparkListener(listener)
+  sparkContext.ui.foreach(new SQLTab(this, _))
```

Comment:
What if I create multiple SQLContexts for the same SparkContext?  Will I get multiple tabs?